src/git-repository-provider.coffee

```diff
@@ -0,0 +1,80 @@
+fs = require 'fs'
+GitRepository = require './git-repository'
+
+# Checks whether the specified directory has, or has a parent directory that
```

Comment:
Could we invert the phrasing for clarity: "Checks whether a valid `.git` directory is contained within the given directory or one of its ancestors." Just a thought. Not a huge deal.